interfaces.d.ts

```diff
@@ -31,6 +31,47 @@ export interface PolymerInit {
   listeners?: {[key: string]: string};
 }
 
+/**
+ * The object passed to ".*" wildcard obsevers. A record of a change made to an
+ * object.
+ * @template B The object matching the non-wildcard portion of the path.
+ * @template V Additional types that could be set at the path.
+ */
+export interface PolymerDeepPropertyChange<B, V> {
+  /** Path to the property that changed. */
+  path: string;
+  /** The object matching the non-wildcard portion of the path. */
+  base: B;
+  /** New value of the path that changed. */
+  value: B|V;
+}
+
+/**
+ * A record of changes made to an array.
+ * @template T The type of the array being observed.
+ */
+export interface PolymerSplice<T extends Array<{}>> {
```

Comment:
Prefer `unknown` over `{}` as `{}` is non-nullable. `T extends Array<unknown>`

Reply:
`unknown` isn't in g3 yet, how about `{}|null|undefined` for now? I think `any` is actually safe here too, because I think in this context (constraining by supertype), the `any` wouldn't leak anywhere (but I would avoid it just so linters don't complain)?

Reply:
{}|null|undefined is good.

You're right about `any`, I'd just like to avoid using it as much as possible.

Reply:
Done.